Fixes #6855. mod_search. Option *Box Width* (size) ignored if INPUT width also set by CSS.

bertmert opened the pull request:
See discussion here
https://github.com/joomla/joomla-cms/issues/6855

Test:
- Use template Protostar.
- Create new module mod_search.
- Enter 10 in field _Box Width_ (size).
- See front end: Size attribute of search input is ignored because Bootstrap CSS sets a fixed width that overrides size attribute.
- Apply patch.
- See front end: Search INPUT has width (size) 10 now.
- Leave field _Box Width_ empty in module settings.
- CSS width rules now.

bertmert committed:
Reset width (CSS) if attribute size given

bertmert committed:
Remove default for size attribute (width)

bertmert committed:
Remove default for size attribute (width)

bertmert committed:
Update default.php

bertmert committed:
Update default.php

Webdongle commented:
@test

Applied patch
Changed size of search box in the search module edit screen
Viewed from front end No Avail
No change in the size of the search box

Does Not work<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6873">issues.joomla.org/joomla-cms/6873</a>.</sub>

bertmert commented:
@Webdongle Cannot confirm.
Installed a fresh Joomla 3.4.1 and got these results:

Before patch

![mod_search](https://cloud.githubusercontent.com/assets/11038612/7424825/eb642d10-efa6-11e4-8e05-455825dd440e.jpg)

After patch

![mod_search2](https://cloud.githubusercontent.com/assets/11038612/7424828/fb3de802-efa6-11e4-873e-713da9c93d16.jpg)

Test with Firebug after patch

![mod_search3](https://cloud.githubusercontent.com/assets/11038612/7424830/0b278142-efa7-11e4-8d34-c0701a21c0b6.jpg)

bertmert committed:
Just code cleaning for Travis

infograf768 commented:
This works here.

Webdongle commented:
Tested again this time success ... the width of the box changed

SUCCESS

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6873">issues.joomla.org/joomla-cms/6873</a>.</sub>

zero-24 commented:
Great so RTC'ing here :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6873">issues.joomla.org/joomla-cms/6873</a>.</sub>

bertmert commented:
@Webdongle 
> The search box in the components area of the results page does not alter

Component has no setting *Box Width*. 
size=30 is hard coded there.

So, this argument does not convince me concerning component com_search
https://github.com/joomla/joomla-cms/issues/6855#issuecomment-97604294

bertmert committed:
Update default.php

Line 20:
$doc is not always defined. So, changed $doc to JFactory::getDocument()

Webdongle commented:
On 01/05/2015 15:43, bertmert wrote:
>
> @Webdongle <https://github.com/Webdongle>
>
>     The search box in the components area of the results page does not
>     alter
>
> Component has no setting /Box Width/.
> size=30 is hard coded there.
>
> So, this argument does not convince me concerning component com_search
>

@bertmert

It was not an argument to convince you of anything ...  it was an 
observation. Your quote of me was out of context.  The full quote "/The 
search box in the components area of the results page does not alter but 
the search box in the module position works as expected/". There are two 
clauses in that sentence with the word 'but' as a conjunction 
http://www.bbc.co.uk/worldservice/learningenglish/grammar/learnit/learnitv161.shtml 
.  That implies the first clause has no direct affect on the second.  So 
please do not twist my words by quoting part of the sentence.

To be clear (just in case English is not you first language) ...
After my first test failed and others tested successfully ... I realised 
that my test must have been inaccurate so I tested again. That test was 
successful but I noticed the search box in the result page did not 
change.  To save time typing several sentences I used one sentence with 
two clauses.  I should have realised that my correct use of English 
Grammar would have been misinterpreted.  And I should have put my 
observation of the results page in a separate paragraph explaining that 
I am not as thick as you think am because I realise they are two 
different things.

bertmert commented:
@Webdongle 

To be clear (just in case you're not able to understand Github principles)

This PR has a title. It's referenced to an issue plus discussion, that also has a title. Read it. 

If you have additional problems with **com_**search, that needs a different way to solve it because com_search has no option *Box Width*, do not use issues concerning **mod_**search because you're comment will be ignored. If it would be a smiliar solution I would have created an additional PR because mbabker's comment would have convinced me.
Thus **YOU** have to open a new issue, provide a test instruction, define a feature request, describe your expected result, whatever ... instead of telling me bla bla... (Be aware that I won't read it because I'm not a native-speaking English teacher and will not understand it.)
